sale_order_import_invoice2data/wizard/sale_order_import_pdf.py

```diff
@@ -0,0 +1,178 @@
+# -*- coding: utf-8 -*-
+# © 2016 Sunflower IT (http://sunflowerweb.nl)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import models, api, tools, _
+from openerp.exceptions import Warning as UserError
+from openerp.tools import float_compare
+from tempfile import mkstemp
+import base64
+import unicodecsv
+import os
+import re
+import pkg_resources
+import logging
+logger = logging.getLogger(__name__)
+
+try:
+    import unicodecsv
+except ImportError:
+    logger.debug('Cannot import unicodecsv')
+
+try:
+    from invoice2data.main import extract_data
+    from invoice2data.template import read_templates
+    from invoice2data.main import logger as loggeri2data
+except ImportError:
+    logger.debug('Cannot import invoice2data')
+
+class SaleOrderImport(models.TransientModel):
+    _inherit = 'sale.order.import'
+
+    # @api.model
+    # def fallback_parse_pdf_saleorder(self, file_data):
+    #     '''This method must be inherited by additionnal modules with
+    #     the same kind of logic as the account_bank_statement_import_*
+    #     modules'''
+    #     return self.invoice2data_parse_saleorder(file_data)
+    #
+    @api.model
+    def parse_pdf_order(self, file_data, detect_doc_type=False):
+        logger.info('Trying to analyze PDF saleorder with invoice2data lib')
+        if detect_doc_type:
+            return 'rfq'
+        fd, file_name = mkstemp()
+        try:
+            os.write(fd, file_data)
+        finally:
+            os.close(fd)
+        # Transfer log level of Odoo to invoice2data
+        loggeri2data.setLevel(logger.getEffectiveLevel())
+        local_templates_dir = tools.config.get(
+            'invoice2data_templates_dir', False)
+        logger.debug(
+            'invoice2data local_templates_dir=%s', local_templates_dir)
+        templates = []
+        if local_templates_dir and os.path.isdir(local_templates_dir):
+            templates += read_templates(local_templates_dir)
+        exclude_built_in_templates = tools.config.get(
+            'invoice2data_exclude_built_in_templates', False)
+        if not exclude_built_in_templates:
+            templates += read_templates(
+                pkg_resources.resource_filename('invoice2data', 'templates'))
+        logger.debug(
+            'Calling invoice2data.extract_data with templates=%s',
+            templates)
+        testspath = os.path.dirname(os.path.realpath(__file__))
```

Comment:
no, not yet. For now, just remove this part of the code and set your template path in the test. Then it's very simple to change this later to use the yet-to-be-written module, and the test template will not interfere with normal operations